hid-xpadneo/src/hid-xpadneo.c

```diff
@@ -899,7 +906,12 @@ static int xpadneo_raw_event(struct hid_device *hdev, struct hid_report *report,
 		bits |= (data[14] & (BIT(0) | BIT(1))) >> 0;	/* A, B */
 		bits |= (data[14] & (BIT(3) | BIT(4))) >> 1;	/* X, Y */
 		bits |= (data[14] & (BIT(6) | BIT(7))) >> 2;	/* LB, RB */
-		bits |= (data[16] & BIT(0)) << 6;	/* Back */
+		if (xdata->quirks & XPADNEO_QUIRK_SHARE_BUTTON) {
```

Comment:
Do we really need a quirk flag for that? What do do the back and share buttons look like in `btmon`?

Reply:
IOW, for my controller, the bit positions of A,B,X,Y,LB,RB,Back.Menu,LS,RS,Xbox are 0,1,3,4,6,7,16,11,13,14,12. Does this match your controller? Do order that correctly by symbolic order, I'm shifting the bits to 0..10, thus share should go to 11. (counted from 0 as in the code)

Reply:
Don't test with games yet at this stage, nor SDL. `evtest` and `jstest` have to work correctly first.

Reply:
We need a quirk flag because the share button is `data[16] & BIT(0)`, the same as the select button on other models. The actual select button is at `data[15] & BIT(2)`, which as far as I can tell is unused by other models.

Reply:
> 7,16,11

And yes, the controller has a button bit in a position that's not covered by the HID descriptor... But only with some firmwares...

Reply:
> We need a quirk flag because the share button is `data[16] & BIT(0)`, the same as the select button on other models. The actual select button is at `data[15] & BIT(2)`, which as far as I can tell is unused by other models.

So MS plays funny games again... `data[16] & BIT(0)` has the back button on SOME firmware versions, on others it's sent in a separate report (opcode 2 instead of 1 I believe) as a consumer control code (AC Back). So, yes: In that case a quirk flag is perfectly suitable.

So in the HID descriptor patcher just make it 12 buttons than - no matter which controller we have. Since we cannot tell the models apart here and it feels wrong looking at the PID at this point, you should move the quirk flag to the `xpadneo_quirks` struct and put an OUI matcher: You model has a different OUI than other models.

Then in the raw event patcher, do not try to play too many tricks: You'll either get the share button at `data[15] & BIT(0)` and move it to bit 11(+1), or you'll get the back button and move it to bit 10(+1). (+1 unless counting from 0). It looks like your back button either comes in report opcode 2, or is at a different position. The code should be clear and easy to understand, I'm currently having problems grasping the purpose of
```c
    bits |= (data[15] & BIT(2)) ? BIT(6) : 0; /* Back */
} else {
    bits |= (data[16] & BIT(0)) << 6; /* Back */
```

Maybe you wanted `bits |= (data[15] & BIT(2)) << 4;` which moves bit 2 of byte 15 to bit 6 of `u16 bits`? Because your back button is at bit 2 byte 15?